File: valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> brackets = new Stack<>();

        for(int i=0; i<s.length(); i++) {
            char c = s.charAt(i);
            if(c == '(' || c == '{' || c == '[') {
                brackets.push(c);
            }
            else {
                if(brackets.size() == 0) {
                    return false;
                }
                char popc = brackets.pop();
                if(c == ')' && popc != '(' ) {
                    return false;
                }
                if(c == '}' && popc != '{' ) {
                    return false;
                }
                if(c == ']' && popc != '[' ) {
                    return false;
                }
            }
        }

        if(brackets.size() > 0) {
            return false;
        }
        return true;
    }
}

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk = []

        for char in s:
            if char == '(' or char == '{' or char == '[':
                stk.append(char)
            else:
                if len(stk) == 0:
                    return False
                char_open = stk.pop()
                if char == ')' and char_open != '(':
                    return False
                elif char == ']' and char_open != '[':
                    return False
                elif char == '}' and char_open != '{':
                    return False

        return len(stk) == 0


        